File: ambeth/Ambeth.Bytecode/ambeth/bytecode/behavior/WaitForApplyBehaviorDelegate.cs
using De.Osthus.Ambeth.Annotation;
using De.Osthus.Ambeth.Bytecode.Visitor;
using De.Osthus.Ambeth.Ioc;
using De.Osthus.Ambeth.Ioc.Annotation;
using De.Osthus.Ambeth.Util;
using System;
using System.Collections.Generic;
using System.Reflection;

namespace De.Osthus.Ambeth.Bytecode.Behavior
{
    public delegate IClassVisitor WaitForApplyBehaviorDelegate(IClassVisitor visitor, IBytecodeBehaviorState state, IList<IBytecodeBehavior> remainingPendingBehaviors, IList<IBytecodeBehavior> cascadePendingBehaviors);
}

File: jambeth/jambeth-bytecode/src/main/java/de/osthus/ambeth/bytecode/behavior/WaitForApplyBehaviorDelegate.java
package de.osthus.ambeth.bytecode.behavior;

import java.util.List;

import de.osthus.ambeth.repackaged.org.objectweb.asm.ClassVisitor;

public interface WaitForApplyBehaviorDelegate
{
	ClassVisitor extend(ClassVisitor visitor, IBytecodeBehaviorState state, List<IBytecodeBehavior> remainingPendingBehaviors,
			List<IBytecodeBehavior> cascadePendingBehaviors);
}
